core/src/main/java/io/questdb/griffin/model/WindowColumn.java

```diff
@@ -203,6 +206,31 @@ public boolean requiresOrderBy() {
                 framingMode == FRAMING_GROUPS;
     }
 
+    public boolean stopOrderByPropagate(ObjList<ExpressionNode> orderByAdvice, IntList orderByDirectionAdvice) {
+        // Range frames work correctly depending on the ORDER BY clause of the subquery, which cannot be removed by the optimizer.
+        boolean stopOrderBy = framingMode == FRAMING_RANGE && !Chars.equalsIgnoreCase(getAst().token, RowNumberFunctionFactory.NAME)
+                && !Chars.equalsIgnoreCase(getAst().token, RankFunctionFactory.NAME) &&
+                orderBy.size() > 0 && ((rowsHi != 0 || rowsLo != Long.MIN_VALUE) && !(rowsHi == Long.MAX_VALUE && rowsLo == Long.MIN_VALUE));
+
+        // Heuristic. If current recordCursor has orderBy column exactly same as orderBy of window frame, we continue to push the order.
+        if (stopOrderBy) {
+            boolean sameOrder = true;
+            if (orderByAdvice.size() != orderBy.size()) {
```

Comment:
I think we can have a weaker condition and allow order by advice to sort by more columns than the window function if they are the same in the beginning.

For example I believe we can propagate the advice in this case:

```SQL
select * from (
  SELECT
      timestamp,
      symbol,
      price,
      first_value(price) OVER (
          partition by symbol
          ORDER BY timestamp ASC
          RANGE BETWEEN 3 seconds preceding and current row
      ) AS first_price
  from 
  ( 
    select * FROM
    read_parquet('trades_nulls.parquet')
    WHERE
        timestamp >= '2024-08-22T14:15:29.206'
    ORDER BY timestamp
  )
)
order by timestamp, symbol
LIMIT 40;
```

The outer sort will sort by timestamp (and then by symbol) but that's ok for the order function, it only needs timestamp sorted.